src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -29,66 +30,113 @@ internal abstract partial class AbstractAddParameterCheckCodeRefactoringProvider
         where TExpressionSyntax : SyntaxNode
         where TBinaryExpressionSyntax : TExpressionSyntax
     {
-        protected override async Task<ImmutableArray<CodeAction>> GetRefactoringsAsync(
-            Document document, IParameterSymbol parameter, SyntaxNode functionDeclaration, IMethodSymbol method,
-            IBlockOperation blockStatementOpt, CancellationToken cancellationToken)
+        protected override async Task<ImmutableArray<CodeAction>> GetRefactoringsForAllParametersAsync(
+            Document document, SyntaxNode functionDeclaration, IMethodSymbol methodSymbol,
+            IBlockOperation blockStatementOpt, IEnumerable<SyntaxNode> listOfParameterNodes, int position, CancellationToken cancellationToken)
         {
-            // Only should provide null-checks for reference types and nullable types.
-            if (!parameter.Type.IsReferenceType &&
-                !parameter.Type.IsNullable())
+
+            if (listOfParameterNodes == null)
             {
                 return ImmutableArray<CodeAction>.Empty;
             }
 
-            var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
-            var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+            // List to keep track of the valid parameters
+            var listOfParametersOrdinals = new List<int>();
 
-            // Look for an existing "if (p == null)" statement, or "p ?? throw" check.  If we already
-            // have one, we don't want to offer to generate a new null check.
-            //
-            // Note: we only check the top level statements of the block.  I think that's sufficient
-            // as this will catch the 90% case, while not being that bad an experience even when 
-            // people do strange things in their constructors.
-            if (blockStatementOpt != null)
+            foreach (var parameterNode in listOfParameterNodes)
             {
-                if (!CanOffer(blockStatementOpt.Syntax))
+                var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+                var parameter = (IParameterSymbol)semanticModel.GetDeclaredSymbol(parameterNode, cancellationToken);
+                if (ParameterValidForNullCheck(document, parameter, semanticModel, blockStatementOpt, cancellationToken))
                 {
-                    return ImmutableArray<CodeAction>.Empty;
+                    listOfParametersOrdinals.Add(parameter.Ordinal);
                 }
+            }
 
-                foreach (var statement in blockStatementOpt.Operations)
+            if (!listOfParametersOrdinals.Any())
+            {
+                return ImmutableArray<CodeAction>.Empty;
+            }
+
+            // Great.  The list has parameters that need null checks. Offer to add null checks for all.
+            return ImmutableArray.Create<CodeAction>(new MyCodeAction(
```

Comment:
shouldn't we only do this if there are at least two parameters that could have null checks?  otherwise, "for all" makes less sense.

Reply:
I was thinking we can save the user the time to check if any exists and just put a null check in even for one parameter - but not a big difference so changed it.

Reply:
actually I think it would be more consistent to add a null check for all even in a case like this:
```cs
foo (int [||]a, string b) {}
```
this would not offer anything if we go with your suggestion. what do you think?

Reply:
However, in this case:
```cs
foo (int a, object [||] b) {}
```
it will only offer to add a single check for one parameter. We only offer one here b/c add null checks for all will be identical to the first option.

Reply:
> this would not offer anything if we go with your suggestion. what do you think?

That makes sense to me.  The person is on an 'int' so i think it's fine to have nothing related to null checking params offered there.

Reply:
The trigger condition in the document (not yet published) states "Place your cursor on any parameter within the method.". So this would violate that. But I don't feel strongly about this.

---
In reply to: [312289520](https://github.com/dotnet/roslyn/pull/36931#discussion_r312289520) [](ancestors = 312289520)

Reply:
For consistency sake it would IMO be preferable if we could fall back to just using common helpers at context.TryGetRelevantNode<Argument> || context.TryGetRelevantNode<ArgumentList> so that the rules for this refactoring and all others are consistent.